### ISSUE TITLE
batdiff: view changes relative to index if no files are passed

### ISSUE THREAD

**Issue by sumanthratna** [eth-p/bat-extras#41]
If I run `batdiff` in my git repository, I get the following output:
```
[batdiff error]: no files provided
```
This is inconsistent with the behavior of `git diff`, which shows the diff of all the modified tracked files.

**Comment by eth-p** [comment]
There is `batdiff --all` which does exactly that, but I suppose it would make more sense to keep it consistent with `git diff`.

**Comment by eth-p** [comment]
I decided to change the default behavior in 0f6529a0d86a65e100042466339e8629c38cb9fe to be more consistent.

**Comment by sumanthratna** [comment]
Thanks—I installed from head and batdiff works as expected. I'll need to play around with paging settings but I'll open another issue (and maybe PR if I have the time) if I have any trouble with it. Just some very, very minor feedback:

when I run `batdiff --all`, I see:
```
[batdiff warning]: argument --all is deprecated. Use '/usr/local/bin/batdiff' instead
the diff
```
If the diff is big, the warning can go unnoticed; it might be more visible if the warning is at the bottom of the diff (although I generally expect program warnings to be before the output). Just pointing this out; no strong opinions.

also: in the deprecating warning the abspath to batdiff is printed—maybe just print the name of the executable file? also not a very important issue.

Closing since you addressed my feature request; thanks again!
